src/Platform/Microsoft.Testing.Platform/CommandLine/ParseResult.cs

```diff
@@ -89,15 +89,15 @@ public bool IsOptionSet(string optionName)
 
     public bool TryGetOptionArgumentList(string optionName, [NotNullWhen(true)] out string[]? arguments)
     {
-        arguments = null;
-
-        IEnumerable<OptionRecord>? result = Options.Where(x => x.Option == optionName.Trim(OptionPrefix));
-        if (result?.Any() == true)
+        optionName = optionName.Trim(OptionPrefix);
+        var result = Options.Where(x => x.Option == optionName).ToList();
+        if (result.Count != 0)
```

Comment:
The `ToList` call is probably not a good idea, as it forces the creation of the collection to then discard it (check for any) by processing another `SelectMany`.

```suggestion
        var result = Options.Where(x => x.Option == optionName);
        if (!result.Any())
```